### ISSUE TITLE
multi gpus training results in zigzag loss curve 

### ISSUE THREAD

**Issue by zdgithub** [horovod/horovod#1527]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) tensorflow
2. Framework version: 1.13.1
3. Horovod version: 0.18.1
4. MPI version:4.0.0
5. CUDA version:10.0
6. NCCL version:2.4.8
7. Python version:3.6
8. OS and version:Ubuntu 18.04
9. GCC version:4.8.5

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here.
when I trained my network on multiple gpus by horovod, I got the following loss curve, which was zigzag:
![loss2](https://user-images.githubusercontent.com/24309565/69250222-3d7bd580-0bea-11ea-810f-5c30ee6a6bd7.png)
But when I trained it on a single gpu without horovod, the loss would decrease smoothly and they had the same initial learning rate:
![lsoss2](https://user-images.githubusercontent.com/24309565/69250238-466ca700-0bea-11ea-8a36-999ae02b9235.png)
Because training on a single gpu is very slow, I just trained 30 epochs that is less than 50 epochs.

[Here](https://discuss.pytorch.org/t/strange-behavior-with-sgd-momentum-training/7442/2) someone thought it might be about the dataset into the network in each epoch, so I wonder if there is anything to do with the dataset on distributed works.

**Comment by tgaddair** [comment]
Hey @zdgithub, how many GPUs are you using to train?  For very large batch sizes, model convergence can start to break down.

@jaliyae, @klipto, @Tixxx, and some other folks from Microsoft recently added a new gradient aggregation method called Adasum in #1485.  If you're having issues with convergence as you scale up the GPUs, you may want to give their approach a shot.

**Comment by zdgithub** [comment]
Thanks for your reply. I use 3 gpus and batch_size is 128.
